.github/workflows/label-analysis.yml

```diff
@@ -46,6 +46,38 @@ jobs:
           exit 1
         env:
           LABELS: ${{ needs.fetch-labels.outputs.LABELS}}
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r tasks/requirements.txt
+      - name: Add get-changed-files dependency
+        id: files
+        uses: jitterbit/get-changed-files@v1
+        with:
+          format: 'csv'
+      - name: Auto assign team label
+        run: |
+          command=( inv -e github.deduce-team-label )
+          mapfile -d ',' -t files < <(printf '%s,' '${{ steps.files.outputs.all }}')
+          for file in "${files[@]}"; do
+            command+=( --changed-files="$file" )
+          done
+          echo "COMMAND: ${command[@]}"
+          labelname="$("${command[@]}")"
+          exitcode="$?"
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏I'm not a big fan of having the github actions be half python half bash, I find the overall step to be hard to read (you need to context-switch between Python and Bash). Maybe this part could be done in the Python as well, WDYT?